pilot/pkg/model/context.go

```diff
@@ -585,7 +585,10 @@ const (
 
 	// NodeMetadataSdsTokenPath specifies the path of the SDS token used by the Envoy proxy.
 	// If not set, Pilot uses the default SDS token path.
-	NodeMetadataSdsTokenPath = "SDS_TOKEN_PATH"
+	NodeMetadataSdsTokenPath = "ISTIO_META_SDS_TOKEN_PATH"
```

Comment:
Why ISTIO_META prefix ? That's used internally to push env variables into metadata - the keys don't include it.

Reply:
we need to put udspath/tokenpath into metadata, so that in pilot lds/cds could use to construct sds.

Reply:
I still don't get it. ISTIO_META is used as a prefix in env variables - there is no env variable involved in this PR. 

Metadata uses norml strings ( and because it is sent on each request / ack / nack - we are hoping to reduce a bit the bandwidth by using shorter strings and only sending what is required ).

Reply:
adding env var will be in following PRs(change on sidecar injector/installer).

The reason to put metadata is, in codepath like [ConstructSdsSecretConfig](https://github.com/istio/istio/blob/master/pilot/pkg/security/model/authentication.go#L115), today we use meshconfig global settings(udspath/tokenpath) to construct sdsconfig; to support migration case, I plan to put udspath/token path in metadata and remove using global meshconfig in above codepaths (in later PRs); wdyt ?

Reply:
The way it works:
- PilotAgent has some special handling for all env starting with ISTIO_META and ISTIO_METAJSON: it strips the prefix and adds them to metadata. Those are generated internally by injector based on annotations and settings. 
- Your code only needs a standard env variable ( not ISTIO_META) to optionally support non-k8s modes where /var/run/  needs to be replaced. Ok to do it in a separate PR or never.
- In any case ISTIO_META is never visible outside injector+agent - Pilot will never get anything starting with ISTIO_META.
- your code is supposed to check for the existence of the well-known SDS path and token. I don't think we need to send the path on each request - again, it's a lot of redundant data that is sent 1000 of times on each connection. If the uds path is a constant in agent - you can use a constant in Pilot, and not send so much data. Mandar is also adding lots of data - it's becoming a perf problem, we need to reduce redundancy.

All this meta is internal to pilot / agent. We could use 'sds=1' or 'sdstoken=1' to indicate the presence of the UDS socket and token. Or use some enum. It's not a huge problem to use longer strings - but sending the full path on every message between agent and pilot is just wasteful. 

On removing the global meshconfigs - not a big priority if they are constants in agent, and there is not a huge reason to make it too flexible, adds complexity with little benefit.
Fine to keep them as meshconfigs for now - or move them to constants in later PR. What really matters is to generate the SdsSecretConfig only if the 'sds' metadata is present. 

The token probably needs to be sent as an id/enum or number - since there are 2 well-known paths. Not the end of the world if we send the full path, just more waste.

Reply:
updated to pass to two flags to metadata(sdsenable/tokenpath),